render/reply/pickle.go

```diff
@@ -115,15 +118,28 @@ func (*Pickle) Reply(w http.ResponseWriter, r *http.Request, multiData data.CHRe
 	}
 
 	nextMetric := data.GroupByMetric()
+	writtenMetrics := make(map[string]struct{})
+	// fill metrics with points
 	for {
 		points := nextMetric()
 		if len(points) == 0 {
 			break
 		}
-		if err := writeMetric(points); err != nil {
+		if err := writeMetric(points, writtenMetrics); err != nil {
 			return
 		}
 	}
+	// fill metrics without points with NaN
+	if multiData[0].AppendOutEmptySeries && len(writtenMetrics) != data.AM.Len() && data.CommonStep > 0 {
+		for _, metricName := range data.AM.Series(false) {
+			if _, done := writtenMetrics[metricName]; done {
```

Comment:
May be simplify to
`if _, done := writtenMetrics[metricName]; !done {`
and avoid of continue block ?